prometheus_client/decorator.py

```diff
@@ -39,6 +39,9 @@
 import operator
 import re
 import sys
+from typing import Any, Callable, TypeVar
```

Comment:
This file is actually a copy of http://pypi.python.org/pypi/decorator, you can see the file header for more details. That does mean we should not edit it so as to make future upgrades easier.

Reply:
I removed it, but out of curiosity. Why is this needed and why contextmanager is used?

For example:

```python
class ExceptionCounter:
    def __init__(self, counter: "Counter", exception: Type[BaseException]) -> None:
        self._counter = counter
        self._exception = exception

    def __enter__(self) -> None:
        pass

    def __exit__(self, typ: Optional[Type[BaseException]], value: Optional[BaseException], traceback: Optional[TracebackType]) -> Literal[False]:
        if isinstance(value, self._exception):
            self._counter.inc()
        return False

    def __call__(self, f: "F") -> "F":
        def wrapped(func, *args, **kwargs):
            with self:
                return func(*args, **kwargs)

        return decorate(f, wrapped)
```

could be:

```python
@contextmanager
def exception_counter(counter: "Counter", exception: Type[BaseException]) -> None:
  yield
  if isinstance(value, self._exception):
    self._counter.inc()
```

Reply:
You can find the history of the vendored in file in the PR that added it, as well as the linked issue: https://github.com/prometheus/client_python/pull/91.

I think you probably could restructure the various context managers like you suggest now that we do not support older python versions. I believe that before 3.2 just using `@contextmanger` would not allow you to use it as a decorator as well so the class was nicer.